libclc/libspirv/lib/ptx-nvidiacl/conversion/GenericCastToPtrExplicit.cl

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <libspirv/spirv.h>
+
+_CLC_DEF static bool __clc_nvvm_is_private(generic void *ptr) {
+  return __nvvm_isspacep_local(ptr);
+}
+_CLC_DEF static bool __clc_nvvm_is_local(generic void *ptr) {
+  return __nvvm_isspacep_shared(ptr);
+}
+_CLC_DEF static bool __clc_nvvm_is_global(generic void *ptr) {
+  return __nvvm_isspacep_global(ptr);
+}
+
+#define GenericCastToPtrExplicit_To(ADDRSPACE, NAME)                           \
+  _CLC_DECL _CLC_OVERLOAD                                                      \
+      ADDRSPACE void *__spirv_GenericCastToPtrExplicit_To##NAME(               \
+          generic void *ptr, int unused) {                                     \
```

Comment:
```suggestion
          generic void *ptr, int) {                                             \
```

Reply:
I tried, but can't do that before C23 :)